MVVM 인터페이스 정의, DIContainer 구현

junu0516 opened the pull request:
- 우선 기본적인 작업만 미리 PR 올려놨습니다.
- 아직 몇 가지 확인이 남아서 커밋 몇 개 더 푸시할 예정입니다. 확인 가능할 때 다시 코멘트 달께요
> 필요 작업
- [x] Coordinator 인터페이스 정의
- [x] View, ViewModel 인터페이스 정의
- [x] DIContainer 인터페이스 정의, 구현체 정의
- [x] 메인 모듈에 MVVMInterface 적용 및 확인

junu0516 committed:
feat: #2 SharedModule -> MVVMInterface로 이름 변경

junu0516 committed:
feat: #2 Coordinator 인터페이스 정의

junu0516 committed:
feat: #2 View, ViewModel 인터페이스 정의

junu0516 committed:
chore: #2 SharedModule에 DIContainer 추가

junu0516 committed:
feat: #2 DIContainer, Inject 타입 구현

junu0516 committed:
refactor: #2 DIContainer을 프로토콜로 변경

junu0516 committed:
refactor: #2 Coordinator 접근제어자 수정

junu0516 committed:
Merge branch 'dev' into chore/#2/PBD-40

junu0516 committed:
chore: #2  Feature 모듈에 MVVMInterface, DIContainer 모듈 의존성 적용되도록 설정 변경

junu0516 committed:
chore: #2 변경된 모듈 설정 확인 위해 FeatureHome의 하위 모듈에 테스트용 임시 코드 적용

junu0516 commented:
@Jeeehee @wnsxor1993 
아직..머지 안된김에 아래 작업을 추가하려고 합니다
- 프로젝트에 SceneDelegate 적용 추가
- Tuist 템플릿 좀 더 직관적으로 알 수 있도록 수정
- 세부 프로젝트별 프레임워크 의존성 dynamic/static 여부 확인 후 재정리
- Bundle ID 확인 후 재조정 필요하면 여기에 논의 요청

wnsxor1993 reviewed (commented):
코멘트 남긴 추가 사항 반영되면 태그주세요. 최대한 빨리 확인하고 Approve 진행하겠습니당^^
추가적으로 해당 PR pull 받은 이후 제가 수정하기로 한 Interface 삭제 부분말고 한 가지 더 요청 내용이 있었던 걸로 기억하는데 다시 알려주실 수 있을까요??ㅠ
@junu0516

junu0516 commented:
@wnsxor1993 
죄송합니다 코멘트 달린걸 모르고 있었네요.. 아마 아래 둘 중 하나였던 것 같아요..
1. Entity 모아놓은 모듈의 경우 (만들 예정이면) 굳이 인터페이스 모듈을 갖추거나, (한 모듈 안에서 추상화한다면) 추상화의 대상으로 간주하지 않아도 될 것 같습니다. 
2. (위의 내용 아니면) Service, Manager 등 이름만 봐서 어떤 역할인지 유추하기 힘들거나, 혹은 어떤 역할을 할 지 정해놓지 않은 모듈이 보이는데 우선 제거하는게 나아보입니다.

junu0516 committed:
refactor: #2 모듈 생성 코드 일부 수정

junu0516 committed:
refactor: #2 코디네이터 형식 수정

- navigationController 속성 제거
- navigationController을 외부에서 생성자 통해 주입받도록 수정

junu0516 committed:
chore: #2 HomePresentation에서 HomeCoordinator 의존성 제거

- 모듈 간 순환참조 발생하기 때문에 우선 제거
- 추후 Coordinator 모듈 추상화 적용 혹은 HomePresentation이 Coordinator 추상타입 접근만을 통해 화면전환 가능한 구조로 변환해야 함

junu0516 committed:
feat: #2 SceneDelegate 추가

junu0516 committed:
feat: #2 임시로 HomeCoordinator을 시작 화면으로 보여주는 flow 적용

junu0516 committed:
feat: #2 코디네이터와 뷰모델 간 delegate 패턴 임시 정의

- HomeCoordinator 모듈에 HomePresentation 모듈 의존성 추가
- delegate 패턴을 통한 뷰모델 및 코디네이터 간 커뮤니케이션 임시로 구현

junu0516 committed:
feat: #2 HomeCoordinator에 임시로 뷰모델 로직 적용

junu0516 committed:
feat: #2 CoordinatorWrapper 타입 정의, Coordinator 인터페이스에 부모 타입을 옵셔널로 재정의

wnsxor1993 reviewed (approved):
고생많으셨습니당
